app/models/miq_server/worker_management/monitor.rb

```diff
@@ -72,42 +67,40 @@ def sync_workers
   end
 
   def clean_worker_records
-    processed_workers = []
+    worker_deleted = false
     miq_workers.each do |w|
       next unless w.is_stopped?
       _log.info("SQL Record for #{w.format_full_log_msg}, Status: [#{w.status}] is being deleted")
-      processed_workers << w
       worker_delete(w.pid)
       w.destroy
+      worker_deleted = true
     end
-    miq_workers.delete(*processed_workers) unless processed_workers.empty?
-    processed_workers.collect(&:id)
+
+    miq_workers.reload if worker_deleted
```

Comment:
👍 I like this better than modifying the in-memory association

Reply:
I want this merged to test if my other change in this area is needed on master... 😉